packages/ui-components-react/src/views/stepView/stepView.tsx

```diff
@@ -0,0 +1,188 @@
+ *   ]}
+ *   onSubmit={handleSubmit((data) => console.log(data))}
+ *   register={register}
+ *   errors={errors}
+ *   isSubmitting={isSubmitting}
+ * />
+ * ```
+ */
+export const StepView = memo(
+	({
+		variant = StepViewVariants.Default,
+		title,
+		description,
+		image,
+		progress,
+		icon,
+		text,
+		action,
+		cardContent,
+		steps,
+		options,
+		fields,
+		checkbox,
+		onSubmit,
+		register,
+		handleSubmit,
+		errors,
+		isSubmitting = false,
+		submitButtonLabel = "Continue",
+		loadingText = "Loading...",
+		className
+	}: StepViewProps): JSX.Element => {
+		// Determine which variant to render
+		const renderVariant = useMemo(() => {
+			// Selection variant
+			if (variant === StepViewVariants.Selection) {
+				return <SelectionVariant options={options} />;
+			}
+
+			// Info variant (explicit)
+			if (variant === StepViewVariants.Info) {
+				return <InfoVariant icon={icon} text={text} action={action} />;
+			}
+
+			// Default form variant
+
+			// For form variants, check fields
+			const safeFields = fields || [];
+			const hasValidFields = Array.isArray(safeFields) && safeFields.length > 0;
+			const hasFormProps = register && handleSubmit && onSubmit;
+
+			// Default form variant
+			if (hasValidFields && hasFormProps) {
+				return (
+					<DefaultVariant
+						fields={safeFields}
+						checkbox={checkbox}
+						onSubmit={onSubmit}
+						register={register}
+						handleSubmit={handleSubmit}
+						errors={errors}
+						isSubmitting={isSubmitting}
+						submitButtonLabel={submitButtonLabel}
+						loadingText={loadingText}
+					/>
+				);
+			}
+
+			// Fallback to info variant if form props are missing or no fields
+			return <InfoVariant icon={icon} text={text} action={action} />;
+		}, [
+			variant,
+			fields,
+			icon,
+			text,
+			action,
+			options,
+			checkbox,
+			onSubmit,
+			register,
+			handleSubmit,
+			errors,
+			isSubmitting,
+			submitButtonLabel,
+			loadingText
+		]);
+
+		const imageColumnStyle = { width: "450px", height: "600px" };
+		const imageStyle = {
+			width: "450px",
+			height: "600px",
+			objectFit: "cover" as const,
+			objectPosition: "left top",
+			display: "block" as const
+		};
+
+		return (
+			<div className={`h-full w-full ${className || ""}`}>
+				<div className="flex h-full w-full overflow-y-auto">
+					{/* Image column - hidden on mobile, visible on desktop */}
+					<div
+						className="relative hidden flex-shrink-0 overflow-hidden md:block"
+						style={imageColumnStyle}
+					>
+						{typeof image === "string" ? (
+							<img
+								src={image}
+								alt={title}
+								className="object-cover object-left-top"
+								style={imageStyle}
+							/>
+						) : (
+							<div style={imageColumnStyle}>{image}</div>
+						)}
+					</div>
+
+					{/* Content column */}
+					<div className="relative z-10 min-w-0 flex-1 py-8 pr-8 md:pl-24">
+						<div className="w-full" style={{ maxWidth: "670px" }}>
```

Comment:
no need for extra padding at the top. and let's use tailwindcss classes for consistency
```suggestion
					<div className="flex-1 pb-8 pr-8 md:pl-24">
						<div className="w-full max-w-[670px]">
```